torchx/lib/torchx/backend.ex

```diff
@@ -861,16 +861,62 @@ defmodule Torchx.Backend do
     {left, right} = maybe_upcast(l, r)
     {left_tx, right_tx} = maybe_broadcast_bin_args(out.shape, left, right)
 
-    mod_fun =
-      case l.type do
-        {:u, s} when s in [16, 32, 64] -> &Torchx.remainder/2
-        _ -> &Torchx.fmod/2
-      end
+    {device, _} = left_tx
+
+    if l.type == {:u, 64} do
+      # We emulate u64 numbers with s64.
+
+      # Numbers smaller than max_s64 are kept as positive s64 and fmod
+      # works fine for those.
+
+      reminder_from_positive = Torchx.fmod(left_tx, right_tx)
+
+      # Numbers bigger than max_s64 are kept as negative s64. Consider
+      # such s64 number denoted as x. We can decompose x into two
+      # positive s64 numbers:
+      #
+      #   x = max_s64 + rest
+      #
+      # We can obtain rest as follows:
+      #
+      #   rest = x - max_s64
+      #
+      # We also know that the following holds:
+      #
+      #   x mod y = ((max_s64 mod y) + (rest mod y)) mod y
+      #
+      # We can safely compute operations on the right hand side using
+      # s64 numbers.
+
+      max_s64_tx =
+        Nx.Constants.max_finite(:s64, backend: Nx.BinaryBackend)
+        |> Nx.to_number()
+        |> Torchx.scalar_tensor(to_torch_type(l.type), device)
```

Comment:
Can't you allocate on Torchx directly with `backend: {Torchx.Backend, device: device}`?

Reply:
I was hoping for a better way :D